src/main/kotlin/RailNetwork.kt

```diff
@@ -0,0 +1,72 @@
+
+class RailNetwork(segments:List<RailSegment>, trains:List<Train>){
+
+    private var simulationStep:Int=1
+    private var segments:List<RailSegment> = segments
+    private var trains:List<Train> = trains
+
+    fun simulateOneStep(){
+
+        for(train:Train in trains){
+
+            var currentSegment:Int = train.getCurrentSegment(simulationStep)
+            if(currentSegment > 0) {
+
+                segments[currentSegment - 1].addTrain(train) // -1 because Segment 1 in the Schedule is here Segment 0 in the Array.
+
+            }
+
+        }
+
+        for(segment:RailSegment in segments){
+
+            var delayed = false
+            if(segment.getTrainCount() > segment.getCapacity()){
+
+                delayed = true
+
+            }
+
+            segment.setTrainsDelayed(delayed)
+
+        }
+
+    }
+
+    fun getNetworkStatus():List<List<String>>{
```

Comment:
Der Status des Netzwerks ist in den Zügen festgehalten, d.h. du kannst ihn da ablesen und direkt in auf die Konsole schreiben. Mach das am besten in `Main` - dann brauchst du nicht diese komplexe `List<List<String>>` anlegen.